### PR TITLE
ET+ random orders generation

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/86986

Add and remove WooCommerce tickets from posts like a pro.

### CODE REVIEW

File: src/Service_Providers/Base.php [hunk 1]
```diff
@@ -66,8 +66,13 @@ public function admin_notices() {
...
 
 		foreach ( $this->get_requisite_plugins() as $plugin ) {
-			$links[] = sprintf( '<a href="%1$s" class="thickbox" title="%2$s">%3$s</a>', esc_attr( $plugin['thickbox_url'] ),
-				esc_attr( $plugin['short_name'] ), esc_html( $plugin['short_name'] ) );
+			if ( isset( $plugin['external_download'] ) && true === $plugin['external_download'] ) {
+				$links[] = sprintf( '<a href="%1$s" title="%2$s" target="_blank">%3$s</a>', esc_attr( $plugin['thickbox_url'] ),
+					esc_attr( $plugin['short_name'] ), esc_html( $plugin['short_name'] ) );
+			} else {
+				$links[] = sprintf( '<a href="%1$s" class="thickbox" title="%2$s">%3$s</a>', esc_attr( $plugin['thickbox_url'] ),
+					esc_attr( $plugin['short_name'] ), esc_html( $plugin['short_name'] ) );
```
Comment (ggwicz) [inline src/Service_Providers/Base.php]: you can break this and the above `sprintf()` into multiple lines -- i think you should

File: src/Service_Providers/Tickets_Plus.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			WP_CLI::add_command( 'event-tickets-plus', $this->container->make( 'Tribe__Cli__Tickets_Plus__Command' ) );
+		}
+
+		// avoid sending emails for fake orders
+		add_filter( 'woocommerce_email_classes', array( $this, 'avoid_sending_emails' ), 999 );
+	}
+
+	public function avoid_sending_emails( $classes ) {
```
Comment (ggwicz) [inline src/Service_Providers/Tickets_Plus.php]: docblock?

File: src/Tickets_Plus/Generator/WooCommerce/CLI.php [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+	/**
+	 * @var \Tribe__Cli__WooCommerce__Orders_Generator
+	 */
+	protected $orders_generator;
+
+	public function __construct( Tribe__Cli__WooCommerce__Orders_Generator $orders_generator ) {
+		$this->orders_generator = $orders_generator;
+	}
+
+	public function generate_orders( array $generator_args = null, array $assoc_args = null ) {
```
Comment (ggwicz) [inline src/Tickets_Plus/Generator/WooCommerce/CLI.php]: docblocks?

**Comment by ggwicz** [comment]
## Questions

- A lot of methods and properties are missing docblocks. *Is this okay?* If not, and we are treating this code like our "normal" plugins code, add all missing docblocks. _(There are many, I did not flag each one because too many.)_
- Of the docblocks that exist, many miss the `@since` param. *Is this okay?* If not, and we are treating this code like our "normal" plugins code, add `@since` to all docblocks.

**Comment by lucatume** [comment]
I think, since we are just starting, that we should stick to our standards and add missing doc blocks and since tags. I will iterate push updates on this branch to address this.
The readme file could use some love too (I think it's now just a copy and paste) but that could be addressed in another PR

**Comment by lucatume** [comment]
Added doc blocks and `@since` tags everywhere and made some tweaks, good for another pass

**Review by ggwicz — APPROVED** [review]
Right on — looks good to me and ready to merge 👍

**Pull request merged**